tawazi/_dag/dag.py

```diff
@@ -413,20 +451,92 @@ def setup(
             target_nodes=target_nodes, exclude_nodes=exclude_nodes, root_nodes=root_nodes
         )
 
-        # 3. remove setup nodes
+        # 3. remove non setup nodes
         graph.remove_nodes_from(
             [node_id for node_id in graph if node_id not in self.graph_ids.setup_nodes]
         )
+        return graph
 
-        # 4. execute the graph and set the results to setup_results
-        _, self.results, _ = execute(
-            exec_nodes=self.exec_nodes,
-            results=self.results,
-            active_nodes=self.actives,
-            max_concurrency=self.max_concurrency,
-            graph=graph,
+    def config_from_dict(self, config: Dict[str, Any]) -> None:
+        """Allows reconfiguring the parameters of the nodes from a dictionary.
+
+        Args:
+            config (Dict[str, Any]): the dictionary containing the config
+                example: {"nodes": {"a": {"priority": 3, "is_sequential": True}}, "max_concurrency": 3}
+
+        Raises:
+            ValueError: if two nodes are configured by the provided config (which is ambiguous)
+        """
+
+        def node_values(n: ExecNode, conf: Dict[str, Any]) -> Dict[str, Any]:
+            values = dataclasses.asdict(n)
+            values["args"] = n.args
+            values["kwargs"] = n.kwargs
+            values["is_sequential"] = conf.get("is_sequential", n.is_sequential)
+            values["priority"] = conf.get("priority", n.priority)
+            return values  # ignore: typing[no-any-return]
+
+        def expand_config(
+            config_nodes: Dict[Union[Tag, Identifier], Any]
+        ) -> List[Tuple[Identifier, Any]]:
+            expanded_config = []
+            for alias, conf_node in config_nodes.items():
+                ids = self.alias_to_ids(alias)
+                expanded_config.extend(
+                    [(id, conf) for id, conf in zip(ids, len(ids) * [conf_node])]
+                )
+            return expanded_config
+
+        def detect_duplicates(expanded_config: List[Tuple[Identifier, Any]]) -> None:
```

Comment:
this could probably be a helper functions in a separate file

Reply:
I put it dag.py for the moment. It might be a good idea to put it in contructor.py